queue/abstract.lua

```diff
@@ -75,6 +75,12 @@ local function tube_release_all_tasks(tube)
         return
     end
 
+    -- Custom drivers may not have an implementation of
+    -- the "tasks_by_state" method.
+    if tube.raw['tasks_by_state'] == nil then
+        return
+    end
```

Comment:
I'm against allowing a driver to leave the method non-implemented. queue module is unable to work correctly without this method.

If it would about, say, statistics and queue module would able to work correctly with reduced functionality, then it would be okay to make the method optional. But it is not our case.

I vote to give a meaningful error and keep the method be required. I hope pushing developers to do small extra work to support the new queue release is worthful considering that the result will be correctness of task states.

Reply:
IMHO, it is more friendly to let users continue to work as they did, if possible. But ok, I will move it to required_driver_methods.

Reply:
Updated

Reply:
Thanks!